src/parsers/urdf/geometry.cpp

```diff
@@ -165,9 +165,18 @@ namespace pinocchio
           bool convex = tree.isMeshConvex (linkName, geomName);
           if (convex) {
             bvh->buildConvexRepresentation (false);
+#if HPP_FCL_MAJOR_VERSION < 2
             geometry = bvh->convex;
-          } else
+#else
+            geometry = boost::shared_ptr<fcl::CollisionGeometry>(bvh->convex.get(), [bvh](...) mutable { bvh->convex.reset(); });
```

Comment:
Why don't we use on `hpp::fcl::shared_ptr` ?

Reply:
Because it will require an API change on Pinocchio side - `boost::shared_ptr<fcl::CollisionGeometry>` is part of the defined API of `retrieveGeometry`

Reply:
Does it make sense to do this change for pinocchio v3 ?

Reply:
I think pinocchio v3 should required hpp-fcl >= v2

Reply:
> Does it make sense to do this change for pinocchio v3 ?

Quick answer: no.